lab05/src/mc322/lab05/Dama.java

```diff
@@ -18,15 +18,30 @@ public char presentPiece() {
     		return 'P';
     }
 
-    boolean isValid(){
+    boolean isValid(int []source, int []target, boolean comer){
         boolean valid = false;
-
+        boolean compatible;
+        int y, x;
+        y = Math.abs(source[0] - target[0]);
+        x = Math.abs(source[1] - target[1]);
+        compatible = isCompatible(y, x);
+        if(compatible == false){
+            return false;
+        }else{
+            if(comer == true){
+                valid = true;
+            }else if(comer == false){
+                valid = true;

```

Comment:
Em ambos os casos, comer == true e comer == false, a flag valid recebe true. Então, não faz sentido manter esses if statements. Tudo isso poderia ser substituído por 
```suggestion
        }else{
            valid = true;
        }

```